client/src/pages/ManageBookings/ManageBookings.tsx

```diff
@@ -0,0 +1,98 @@
+import { Box, Grid, Paper, Typography, useTheme } from '@material-ui/core';
+import { DatePicker, MuiPickersUtilsProvider } from '@material-ui/pickers';
+import { MaterialUiPickersDate } from '@material-ui/pickers/typings/date';
+import { cloneElement, useState } from 'react';
+import { CurrentBookings, NextBooking, PastBookings } from '../../components/ManageBookings/';
+import { checkIfBooked } from '../../helpers/ManageBookings';
+import mockBookingData from './mockBookingData';
+import { BookingRequests } from './types';
+import useStyles from './useStyles';
+
+export default function ManageBookings(): JSX.Element {
+  const classes = useStyles();
+  const theme = useTheme();
+  const [date, setDate] = useState<Date | null>(new Date());
+  const [sitterBookings, setSitterBookings] = useState<BookingRequests>(mockBookingData);
+
+  const updateDate = (date: MaterialUiPickersDate) => {
+    setDate(date);
+  };
+
+  const renderDay = (
+    day: MaterialUiPickersDate,
+    selectedDate: MaterialUiPickersDate,
+    dayInCurrentMonth: boolean,
+    dayComponent: JSX.Element,
+  ) => {
+    const renderDay = day?.getTime();
+    const selectedDay = selectedDate?.getTime();
+    const isSelectedDay = selectedDay === renderDay;
+
+    const isBookedDay = checkIfBooked(sitterBookings, day);
+
+    const dayStyle = {
+      style: { backgroundColor: theme.palette.common.white, border: 'none', color: theme.palette.common.black },
+      disabled: true,
+    };
+
+    if (isBookedDay) {
+      dayStyle.style.backgroundColor = theme.palette.primary.main;
+      dayStyle.disabled = false;
+    }
+    if (isSelectedDay) {
+      dayStyle.style.border = `2px solid ${theme.palette.grey[400]}`;
+    }
+    const dayComponentStyled = cloneElement(dayComponent, dayStyle);
+    return <div>{dayComponentStyled}</div>;
+  };
+
+  return (
+    <Grid container component="main" spacing={5} className={classes.root + ' ' + classes.myBookings}>
+      <Grid item sm={6}>
+        <Grid container spacing={2} className={classes.bookingListingContainer}>
+          <Grid item>
+            <Paper elevation={2}>
+              <Box p={2} pt={2}>
+                <Typography variant="h6" component="div" className={classes.uppercase}>
+                  Your Next Booking
+                </Typography>
+                <NextBooking bookingRequests={sitterBookings} />
+              </Box>
+            </Paper>
+          </Grid>
+          <Grid item>
+            <Paper elevation={2}>
+              <Box p={2} pt={2}>
+                <Typography variant="h6" component="div" className={classes.uppercase}>
+                  Current Bookings
+                </Typography>
+                <CurrentBookings bookingRequests={sitterBookings} />
+                <Typography variant="h6" component="div" className={classes.uppercase}>
+                  Past Bookings
+                </Typography>
+                <PastBookings bookingRequests={sitterBookings} />
```

Comment:
We could have been more [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) here, both of these sections are almost identical. If you take a closer look, `PastBookings` and `CurrentBookings` components are exactly the same in representation (e.g. they return the same JSX) and when we end up moving the logic for processing bookings, we will end up with duplicate components. Instead we could componentize these sections by taking a list of bookings, a section title and a fallback (e.g. you have this in `NextBooking` ([`lines 38-40`](https://github.com/hatchways/team-gingerbread/blob/232ca0d94fc825c32cce29d44428ff9e0e560b21/client/src/components/ManageBookings/NextBooking.tsx#L38-L40)), I think it would be nice to have them for `Past/CurrentBookings` as well).

Reply:
I see, that makes a lot of sense.  The processing can happen in the high level component, and then all 3 display components can be instances of the same lower-level display component.